apiserver/observer/metricobserver/metricobserver.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package metricobserver
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils/clock"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/juju/juju/apiserver/observer"
+	"github.com/juju/juju/rpc"
+)
+
+const (
+	facadeLabelIndex = iota
```

Comment:
I'm probably missing something but why do these exist? Could you just define the metric labels as const strings and then define `metricLabelNames` as a list of those?